tools/make-sysext

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+
+# Build and install tree as systemd-sysext.
+# See ../HACKING.MD section "Working on your local Fedora machine: systemd-sysext"
+
+set -eu
+
+SYSEXT_NAME="cockpit-git"
+SYSEXT_DIR="/run/extensions/$SYSEXT_NAME"
+
+if [ "${1:-}" = install ]; then
+    if systemd-sysext --no-legend status | grep -qw "$SYSEXT_NAME"; then
+        systemctl stop cockpit.socket
+        systemd-sysext unmerge
+    fi
+    if [ -d $SYSEXT_DIR ]; then
+        rm -rf "$SYSEXT_DIR"
+    fi
+
+    cp -r tmp/sysext "$SYSEXT_DIR"
+    systemd-sysext merge --force
+
+    echo
+    systemd-sysext status --no-pager
+
+    systemctl start cockpit.socket
+    exit 0
+fi
+
+. /run/host/etc/os-release
+if [ "$ID" != "fedora" ]; then
```

Comment:
Yes, correct. If you *don't* run this in the cockpit/tasks container, you can of course do this for any OS. This restriction is mostly because (1) that PAM file, and (2) I only tested it on Fedora.

Reply:
We can probably generalize this quite easily by comparing /etc/os-release's `$ID` with /run/host/etc/os-release. And if /run/host isn't there, we simply allow it, and assume the developer knows what they are doing? After all, the worst possible result is that you fail with a linker error or so.

Reply:
Done. It drops `--force` and installs a proper compatibility declaration file.